Added the changes to run the find object function

szabipal opened the pull request.

Commit:
Added the changes to run the find object function

Commit:
Added the changes to run the find object function

krisstallenberg committed:
Merge branch 'main' into navigation-b

krisstallenberg committed:
Complete step to clarify the initial description

krisstallenberg committed:
Complete step to clarify the initial description

krisstallenberg commented on AI2ThorClient.py:
```diff
@@ -167,12 +177,89 @@ def parse_unstructured_description(self, description: str):
+        If the target is not found after 4 rotations, teleport to a random location.
+        Args:
+            target (str): Name of the target object.
+            described (list): List of described objects to assist in finding the target.
```
The function signature and docstring don't match

krisstallenberg commented on AI2ThorClient.py:
```diff
@@ -167,12 +177,89 @@ def parse_unstructured_description(self, description: str):
+            described (list): List of described objects to assist in finding the target.
+        Returns:
+            bool: True if the target object is in sight.
+            string: Type of object in visible distance.
```
I don't think you need both. If there is no string, there is no bool and likewise, right?

Also, we may have to rename this function and change the logic when setting the visibility to 10+ meter.

krisstallenberg commented on AI2ThorClient.py:
```diff
@@ -167,12 +177,89 @@ def parse_unstructured_description(self, description: str):
         self.structured_initial_description = response.choices[0].message.parsed
 
+    def _find_objects_and_angles(self, image, target_objects):
+        """Main workflow to find objects and compute turn angles."""
```
```suggestion
        """
        Compute the angle of rotation towards a given object.
        
        Args:
            image: @szabipal populate this
            target_objects: @szabipal populate this
        Returns: 
            turn_angle: the degrees of rotation to face the target object
            detections_df: @szabipal populate this
            selected_objects: @szabipal populate this
        
        
        Main workflow to find objects and compute turn angles.
        """
```

krisstallenberg committed:
Fix issue regarding renamed room_type

krisstallenberg committed:
Update main.py

krisstallenberg committed:
Fix teleport bug due to non-zero x and z rotation

krisstallenberg committed:
Add Thor client method to find nearest room center

To teleport to nearest center of room:

Call AI2ThorClient._teleport(position=AI2ThorClient._find_nearest_center_of_room())

krisstallenberg committed:
Add support for developer mode (#14)

krisstallenberg committed:
Fix bug in finding objects in sight

szabipal committed:
Add navigation logic, find object and approach

szabipal committed:
Add implementation and debugging for the navigation. It takes finds the object for now and references it to a visible object. The navigation to object functions do not work yet.

szabipal committed:
Navigation improvements
